hawk/src/main/java/com/orhanobut/hawk/Hawk.java

```diff
@@ -47,14 +49,10 @@ public static void init(Context context, LogLevel logLevel) {
      * @param value is the data that is gonna be saved. Value can be object, list type, primitives
      */
     public static <T> void put(String key, T value) {
-        if (key == null) {
-            throw new NullPointerException("Key cannot be null");
-        }
-        if (value == null) {
-            throw new NullPointerException("Value cannot be null");
-        }
-        String cipherText = encoder.encode(value);
-        String fullText = DataUtil.addType(cipherText, value.getClass(), false);
+        throwExceptionIf(key == null, "Key cannot be null");
```

Comment:
this method looks really nice but I think it won't be suitable for every case and the method itself doesn't specify which exception exactly it will throw and in some cases we throw different exceptions. I think we should keep the current usage as it is for the validations.